ui/packages/catalog-ui-search/src/main/webapp/component/search-form/forms-sharing/search-form-sharing.collection.js

```diff
@@ -12,109 +12,112 @@
 
- let sharedTemplates = [];
- let promiseIsResolved = false;
- const sharedSearchFormPromise = () => $.ajax({
-    type: 'GET',
-    context: this,
-    url: './internal/forms/query',
-    contentType: 'application/json',
-    success: function (data) {
-        sharedTemplates = data;
-        promiseIsResolved = true;
-    }
-});
+let sharedTemplates = []
+let promiseIsResolved = false
+const sharedSearchFormPromise = () => $.ajax({
+  type: 'GET',
+  context: this,
+  url: './internal/forms/query',
+  contentType: 'application/json',
+  success: function (data) {
+    sharedTemplates = data
+    promiseIsResolved = true
+  }
+})
 
-let bootstrapPromise = sharedSearchFormPromise();
+let bootstrapPromise = sharedSearchFormPromise()
 
- module.exports = Backbone.AssociatedModel.extend({
-   model: SearchForm,
-   defaults: {
-        doneLoading: false,
-        sharedSearchForms: []
-   },
-   initialize: function() {
-       this.addMySharedForms();
-   },
-    relations: [{
-        type: Backbone.Many,
-        key: 'sharedSearchForms',
-        collectionType: Backbone.Collection.extend({
-            model: SearchForm,
-            initialize: function() {}
-        })
-    }],
-   addMySharedForms: function() {
-    if (!this.isDestroyed){
-        if (promiseIsResolved === true) {
-            promiseIsResolved = false;
-            bootstrapPromise = new sharedSearchFormPromise();
-        }
-        bootstrapPromise.then(() => {
-            $.each(sharedTemplates, (index, value) => {
-                if (this.checkIfShareable(value)) {
-                    let utcSeconds = value.created / 1000;
-                    let d = new Date(0);
-                    d.setUTCSeconds(utcSeconds);
-                    this.addSearchForm(new SearchForm({
-                        createdOn: Common.getHumanReadableDate(d),
-                        id: value.id,
-                        name: value.title,
-                        description: value.description,
-                        type: 'custom',
-                        filterTemplate: JSON.stringify(value.filterTemplate),
-                        accessIndividuals: value.accessIndividuals,
-                        accessGroups: value.accessGroups,
-                        createdBy: value.creator,
-                        owner: value.owner,
-                        querySettings: value.querySettings
-                    }));
-                }
-            });
-            this.doneLoading();
+module.exports = Backbone.AssociatedModel.extend({
+  model: SearchForm,
+  defaults: {
+    doneLoading: false,
+    sharedSearchForms: []
+  },
+  initialize: function () {
+    this.addMySharedForms()
+  },
+  relations: [{
+    type: Backbone.Many,
+    key: 'sharedSearchForms',
+    collectionType: Backbone.Collection.extend({
+      model: SearchForm,
+      initialize: function () {}
+    })
+  }],
+  addMySharedForms: function () {
+    if (!this.isDestroyed) {
```

Comment:
✏️ should be able to get rid of the `{}` set and make the code a little more readable if you return early (after reversing this predicate), i.e.
```
if (this.isDestroyed) {
 return;
}
```